### ISSUE TITLE
Numworks Exam Mode

### ISSUE THREAD

**Issue by TLMaths** [numworks/epsilon#1953]
As exam mode doesn’t disable Python - what stops a student from putting their calculator into exam mode early and then typing notes into Python?

The calculator will appear in exam mode (and hence perfectly fine) to invigilators.

We will have 500 students sitting the exam at the same time, so the only solution I can see is to have a computer ready outside of the exam hall where if a calculator is already in exam mode, I can then plug it in, take it out of exam mode, then put it back into exam mode.

Can a future update disable Python for Numworks when in exam mode please?

**Comment by Lisra-git** [comment]
Hello
If the country is set to the correct settings and Python is available in exam mode,
It mean it's perfectly legal for the student to use Python in exam.

About storage, depending on local law, the student should come with the exam mode disabled and will have to enable it in class, but if exam mode is already enabled, 
You can go to settings --> Exam mode to reset Exam mode by deleting all data that may have been hidden while on Exam mode

So no utility to delete Python for Exams 😃

**Comment by TLMaths** [comment]
Thank you for your reply.

I didn’t know about the Exam Mode re-set, so at least that is a quick fix.

Regardless of whether Python is legal in exams, however, I would question why students should have it available at all in Exam Mode. I would push for its disabling, as I can foresee a large logistical challenge for centres like mine to check calculators at the beginning of each calculator-allowed examination.

Thanks again.

**Comment by Lisra-git** [comment]
Python will not get disabled
It's a legal tool for students in exam will local laws doesn't forbid it
Checking each calculator is a legal obligation not some lazy task
If you are appointed to this task, do it fully and don't try to get students to have less tool for your own comfort...

**Comment by Lisra-git** [comment]
What you have to see is 
Python is perfectly a legal tool in UK
Exam mode and usage is all at responsability of the student
Numworks won't adapt to some professor requirements...
Just check other brands such as Casio still have Python will in Exam so Numworks have no point to change it....

**Comment by M4xi1m3** [comment]
Hello,

According to your YouTube channel, you seem to come from the UK. The [guidelines I found](https://www.jcq.org.uk/wp-content/uploads/2022/01/ICE_21-22_FINAL.pdf) on the Internet doesn't forbid the use of Python in exams. Also according to this document, exam mode isn't a requirement at all on these exams, it only says the data must be cleared.

Casio and TI keep Python enabled in exam mode and haven't had problems in years about that.

If you really want to make sure your students clears their data, you have two choices:
 - Either impose exam mode. If they come in the exam room with exam mode, make them re-enable it (same procedure as activating it normally)
 - Don't impose exam mode but make the students reset their calculator, either with the use of the "Reset Calculator" button in the settings or with the use of the reset button which is at the back of the calculator.

Please, for god's sake, don't give Numworks terrible ideas about removing Python from exam mode. We know what they are capable of.

**Comment by parkermaths** [comment]
Casio do disable Python in exam mode.

What possible reason could you have for leaving it enabled? It just makes it more of a logistical challenge for examiners, thereby reducing the uptake of Numworks calculators by schools.

I sincerely hope the Numworks staff themselves don't have such a disdainful attitude towards teaching staff.

**Comment by adriweb** [comment]
> What possible reason could you have for leaving it enabled? 

Many exams are about algorithms and specifically python. It's not quite a coincidence all calculators nowadays have python: curriculums and exams require it.

**Comment by parkermaths** [comment]
There are precisely 0 uk exams that require Python or are specifically about Python algorithms.

There are precisely 0 exams in the UK that require a calculator that can do Python or would even benefit from being able to use Phyton mode for anything other than attempting to cheat.

So, given that this thread is about uk exam mode, my question again is "what possible reason could there be for having it enabled?"

**Comment by adriweb** [comment]
Your question then becomes whether NumWorks can make a UK-specific exam mode (or maybe share it with one that has the same restrictions)

**Comment by hunter1749** [comment]
Instead of disabling everything, it would be nice if there was an option to disable specific apps and features, apart from the ones shown in the exam mode menu.

**Comment by Yaya-Cout** [comment]
It already exists with the Press-to-test mode, non ?

**Comment by parkermaths** [comment]
Not quite. The press to test mode when country is set to international is designed for IB exam, which restricts useful functions that our student can use (e.g. vectors).

The idea behind the press to test mode is similar to what we are trying to achieve, just with the appropriate features disabled as suitable for uk examinations.
